### PR TITLE
Invite contacts via seitan

### PR DESCRIPTION
**Pull request by buoyad**
Almost done, just could use some code cleanup

- [x] Add `isLoading` to check for loading seitan invite tokens (for cancellation waiting state)
- [x] Update copy on opened sms
- [x] Cleanup

### CODE REVIEW

File: shared/actions/teams/index.js [hunk 1]
```diff
@@ -133,42 +133,66 @@ const _editMembership = function*(action: Constants.EditMembership) {
...
+  /* Open SMS */
+  // seitan is 16chars
+  // message sans teamname is 134chars long. Absolute max teamname can be is 26chars to fit within 160 sms limit
+  // limit teamname to 15chars (+3 for ellipsis) to leave some wiggle room
+  let clippedTeamname = teamname
+  if (clippedTeamname.length > 18) {
+    clippedTeamname = `...${teamname.substring(teamname.length - 15)}`
+  }
+  const bodyText = `Please join the ${clippedTeamname} team on Keybase. Install and paste this in the "Teams" tab:\n\ntoken: ${seitan.toUpperCase()}\n\nquick install: keybase.io/_/go`
+  openSMS(phoneNumber, bodyText)
```
Comment (buoyad) [inline shared/actions/teams/index.js]: this has pretty inconsistent behavior on iOS when the messages app isn't running in the background. likely a quirk of iOS's half-support of opening SMS links. can likely be made better by using MessageUI in the future.

File: shared/constants/teams.js [hunk 2]
```diff
@@ -49,14 +49,16 @@ export type InviteToTeamByPhone = NoErrorTypedAction<
...
   }
 >
 
 // username -> removeMember
 // email -> removePendingInvite
+// id -> removePendingSeitanInvite
 export type RemoveMemberOrPendingInvite = NoErrorTypedAction<
   'teams:removeMemberOrPendingInvite',
-  {name: string, username: string, email: string}
+  {name: string, username: string, email: string, id: string}
```
Comment (buoyad) [inline shared/constants/teams.js]: changing `removeMemberOrPendingInvite` to always use `inviteID` for invites would make this typing better, but core support hasn't been thoroughly tested for non-seitan invites (cc @zapu)

File: shared/constants/teams.js [hunk 3]
```diff
@@ -49,14 +49,16 @@ export type InviteToTeamByPhone = NoErrorTypedAction<
...
   }
 >
 
 // username -> removeMember
 // email -> removePendingInvite
+// id -> removePendingSeitanInvite
 export type RemoveMemberOrPendingInvite = NoErrorTypedAction<
   'teams:removeMemberOrPendingInvite',
-  {name: string, username: string, email: string}
+  {name: string, username: string, email: string, inviteID: string}
```
Comment (buoyad) [inline shared/constants/teams.js]: changing `removeMemberOrPendingInvite` to always use `inviteID` for invites would make this typing better, but core support hasn't been thoroughly tested for non-seitan invites (cc @zapu)

File: shared/actions/teams/index.js [hunk 4]
```diff
@@ -135,42 +135,62 @@ const _editMembership = function*(action: Constants.EditMembership) {
...
+  let teamDescription
+  if (teamname.length <= 16) {
+    teamDescription = `${teamname} team`
+  } else {
+    // then this must be a subteam, and won't safely fit into a text
+    const subteams = teamname.split('.')
+    teamDescription = `${subteams[subteams.length - 1]} subteam`
+  }
+  const bodyText = `Please join the ${teamDescription} on Keybase. Install and paste this in the "Teams" tab:\n\ntoken: ${seitan.toUpperCase()}\n\nquick install: keybase.io/_/go`
+  openSMS(phoneNumber, bodyText)
```
Comment (buoyad) [inline shared/actions/teams/index.js]: this has pretty inconsistent behavior on iOS when the messages app isn't running in the background. likely a quirk of iOS's half-support of opening SMS links. can likely be made better by changing `openSMS` to use [MessageUI](https://www.npmjs.com/package/RCTMessageUI) in the future.

**Comment by buoyad** [comment]
@keybase/react-hackers this is ready

**Comment by buoyad** [comment]
@keybase/react-hackers This is ready now! Note: the MessageUI doesn't work on the iOS sim, only on an actual device.

File: shared/react-native/ios/Keybase/AppDelegate.m [hunk 5]
```diff
@@ -125,8 +125,8 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   // that). If you're building onto a phone, you'll have to change
   // localhost:8081 to point to the bundler running on your computer.
   //
-  // jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.ios.bundle?platform=ios&dev=false"];
-  jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index.ios" fallbackResource:nil];
+  jsCodeLocation = [NSURL URLWithString:@"http://10.0.1.19:8081/index.ios.bundle?platform=ios&dev=false"];
```
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/AppDelegate.m]: don't check this in

File: shared/teams/invite-by-email/container.native.js [hunk 6]
```diff
@@ -17,6 +17,19 @@ import {type OwnProps} from './container'
...
 import {getContacts} from './permissions'
 
+const cleanPhoneNumber: string => string = (dirty: string) => {
+  return dirty.replace(/\D/g, '')
+}
+
+// we get invite name as `[name] ([phone number]), this extracts [phone number]
+const extractPhoneNumber: string => ?string = (name: string) => {
+  const matches = /\((.*)\)/.exec(name)
+  if (matches[1]) {
```
Comment (chrisnojima) [inline shared/teams/invite-by-email/container.native.js]: ```matches && ```

File: shared/react-native/ios/Keybase/AppDelegate.m (hunk 5 above)
Comment (buoyad) [reply shared/react-native/ios/Keybase/AppDelegate.m]: Whoops

Comment (buoyad) [reply shared/react-native/ios/Keybase/AppDelegate.m]: whoops!

**Comment by buoyad** [comment]
Addressed comments @chrisnojima, ptal!

File: shared/teams/invite-by-email/container.native.js [hunk 7]
```diff
@@ -24,10 +24,7 @@ const cleanPhoneNumber: string => string = (dirty: string) => {
 // we get invite name as `[name] ([phone number]), this extracts [phone number]
 const extractPhoneNumber: string => ?string = (name: string) => {
   const matches = /\((.*)\)/.exec(name)
-  if (matches[1]) {
-    return cleanPhoneNumber(matches[1])
-  }
-  return ''
+  return (matches[1] && cleanPhoneNumber(matches[1])) || ''
```
Comment (chrisnojima) [inline shared/teams/invite-by-email/container.native.js]: Can't matches itself be null

Comment (buoyad) [reply shared/teams/invite-by-email/container.native.js]: ah yeah, I misunderstood your original comment. Fixed now

**Comment by buoyad** [comment]
Also there seems to be considerable lag on opening the messageUI, I'm not sure what's causing that

**Pull request merged**